packages/aws-rfdk/lib/core/lib/x509-certificate.ts

```diff
@@ -230,6 +274,93 @@ abstract class X509CertificateBase extends Construct {
         StringEquals: tagCondition,
       },
     }));
+
+    this.lambdaMonitor = this.addLambdaMonitorFunction({
+      uniqueValue,
+      lambdaLayers: [openSslLayer],
+      lambdaCode: props.lambdaCode,
+      tagCondition,
+      lambdaMonitorHandler: props.lambdaMonitorHandler,
+    });
+    this.addExpDateValidation({
+      uniqueValue,
+      alarmEmailAddress: props.alarmEmailAddress,
+    });
+  }
+
+  private addLambdaMonitorFunction(props: LambdaMonitorFunctionProps): LambdaFunction{
+    const lambdaMonitor = new LambdaFunction(this, 'MonitoringExpDate', {
+      description: `Used by a X509Certificate ${Names.uniqueId(this)} to monitor certificate expiration date.`,
+      code: props.lambdaCode,
+      environment: {
+        UNIQUEID: props.uniqueValue,
+      },
+      runtime: Runtime.NODEJS_14_X,
+      layers: props.lambdaLayers,
+      handler: props.lambdaMonitorHandler,
+      timeout: Duration.seconds(90),
+      logRetention: RetentionDays.ONE_WEEK,
+    });
+    lambdaMonitor.addToRolePolicy(new PolicyStatement({
+      actions: [
+        'secretsmanager:GetSecretValue',
+      ],
+      resources: ['*'],
+      conditions: {
+        StringEquals: props.tagCondition,
+      },
+    }));
+    lambdaMonitor.addToRolePolicy(new PolicyStatement({
+      actions: [
+        'secretsmanager:ListSecrets',
+      ],
+      resources: ['*'],
+    }));
+    lambdaMonitor.addToRolePolicy(new PolicyStatement({
+      actions: [
+        'cloudwatch:PutMetricData',
+      ],
+      resources: ['*'],
+      conditions: {
+        StringEquals: { 'cloudwatch:namespace': CertificateRotationMonitor.METRIC_NAMESPACE },
+      },
+    }));
+    return lambdaMonitor;
+  }
+
+  private addExpDateValidation(props: X509CertValidationProps){
+    const certMetric = new Metric({
+      metricName: CertificateRotationMonitor.METRIC_NAME,
+      namespace: CertificateRotationMonitor.METRIC_NAMESPACE,
+      dimensions: {},
+      // We're looking at the minimum value reported by the Lambda in a day, which should
+      // be the only value, since it runs once daily.
+      period: Duration.days(1),
+      statistic: 'Minimum',
+    });
+
+    certMetric.dimensions![CertificateRotationMonitor.METRIC_DIMENSION] = props.uniqueValue;
+
+    const alarm = certMetric.createAlarm(this, 'CertificateExpiryAlarm', {
+      evaluationPeriods: 1,
+      threshold: 15,
+      comparisonOperator: ComparisonOperator.LESS_THAN_THRESHOLD,
+      treatMissingData: TreatMissingData.NOT_BREACHING,
+      actionsEnabled: props.alarmEmailAddress ? true : false,
+    });
```

Comment:
Can we add an `alarmDescription` here? I suspect this description will be included in the email notification. We should probably provide or link to the action that the recipient should take to rotate the certificates. See the following example in our example app for setting up similar metric/alarm/notification for EFS burst credits:

https://github.com/aws/aws-rfdk/blob/0933a9e5a569d0f03a266503bfdf59206a2ce41d/examples/deadline/All-In-AWS-Infrastructure-Basic/ts/lib/storage-tier.ts#L200-L285